### DISCUSSION TITLE
android PAGImageView  崩溃

### DISCUSSION THREAD

**Discussion by zengke328** [Tencent/libpag#2277] [Q&A] [answered]
版本号：4.3.47

报错堆栈：
**[libpag.so] pag::PAGSurface::draw(pag::RenderCache*, std::__ndk1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool)**
backtrace:
backtrace:
  #00  pc 0x0000000000062c74  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #01  pc 0x00000000000cb138  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #02  pc 0x00000000000d0950  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #03  pc 0x00000000000cf99c  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #04  pc 0x00000000000cef34  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #05  pc 0x00000000000cf478  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #06  pc 0x00000000000d04d4  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #07  pc 0x00000000000cf99c  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #08  pc 0x00000000000cef34  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #09  pc 0x00000000000d04d4  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #10  pc 0x00000000000cf99c  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #11  pc 0x00000000000a37dc  /data/app/~~/lib/arm64/libpag.so (pag::PAGSurface::draw(pag::RenderCache*, std::__ndk1::shared_ptr<pag::Graphic>, pag::BackendSemaphore*, bool)+464) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #12  pc 0x00000000000a1bd0  /data/app/~~/lib/arm64/libpag.so (pag::PAGPlayer::flushInternal(pag::BackendSemaphore*)+192) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #13  pc 0x00000000000a1ad8  /data/app/~~/lib/arm64/libpag.so (pag::PAGPlayer::flush()+88) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #14  pc 0x000000000009475c  /data/app/~~/lib/arm64/libpag.so (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #15  pc 0x000000000009f3e8  /data/app/~~/lib/arm64/libpag.so (pag::PAGDecoder::renderFrame(std::__ndk1::shared_ptr<pag::PAGComposition>, int, std::__ndk1::shared_ptr<pag::BitmapBuffer>)+516) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #16  pc 0x000000000009eca4  /data/app/~~/lib/arm64/libpag.so (pag::PAGDecoder::readFrameInternal(int, std::__ndk1::shared_ptr<pag::BitmapBuffer>)+680) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #17  pc 0x000000000009ef34  /data/app/~~/lib/arm64/libpag.so (pag::PAGDecoder::readFrame(int, AHardwareBuffer*)+108) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #18  pc 0x0000000000105fdc  /data/app/~~/lib/arm64/libpag.so (Java_org_libpag_PAGDecoder_readFrame+112) (BuildId: c43d6dd0367e51b4744d4c7eafb275d4782fb666)
  #19  pc 0x00000000003d27a4  /data/app/~~/oat/arm64/base.odex (art_jni_trampoline+132)
  #20  pc 0x0000000000e5ede8  /data/app/~~/oat/arm64/base.odex (org.libpag.c$a.a+104)
  #21  pc 0x0000000000ed8648  /data/app/~~/oat/arm64/base.odex (org.libpag.PAGImageView.a+1384)
  #22  pc 0x0000000000ed94a8  /data/app/~~/oat/arm64/base.odex (org.libpag.PAGImageView.flush+312)
  #23  pc 0x0000000000ed9b2c  /data/app/~~/oat/arm64/base.odex (org.libpag.PAGImageView.onAnimationUpdate+156)
  #24  pc 0x0000000000e5ddb0  /data/app/~~/oat/arm64/base.odex (org.libpag.PAGAnimator.onAnimationUpdate+160)
  #25  pc 0x000000000033b3a4  /apex/com.android.art/lib64/libart.so (art_quick_invoke_stub+612)

**Comment by zengke328** [comment]
现在这个crash在逐步升高，帮忙看看

**Reply by kevingpqi123** [reply, marked as answer]
确认下使用场景，是否存在一个页面中创建多个 PAGView 播放同一个 pag 文件？
如果存在，可以更新至 4.3.54

**Reply by zengke328** [reply]
感谢，另外请教下 https://github.com/Tencent/libpag/discussions/1589 这个bug v4.3.54版本有修复么。

**Reply by kevingpqi123** [reply]
crash 的问题你们最好提供使用方式是啥，上面的堆栈感觉有些异常，比如这类反馈 https://github.com/Tencent/libpag/discussions/2272 ，肯定可以定位到原因
